docs/articles/concepts/kafka.md

```diff
@@ -0,0 +1,167 @@
+Bytewax ships with connectors for [Kafka](https://www.confluent.io/) and
+Kafka compatible systems, like [Redpanda](https://redpanda.com/).
+
+In this section, we'll discuss these connectors in detail, as well as
+provide some important operational information about using them as sources
+and sinks.
+
+## Connecting to Kafka
+
+Bytewax provides two basic ways to connect to Kafka.
+
+You can use [`KafkaSource`](/apidocs/bytewax.connectors/kafka/index#bytewax.connectors.kafka.KafkaSource)
+and [`KafkaSink`](/apidocs/bytewax.connectors/kafka/index#bytewax.connectors.kafka.KafkaSink) directly:
+
+```python
+from bytewax.connectors.kafka import KafkaSource, KafkaSink, KafkaSinkMessage
+from bytewax import operators as op
+from bytewax.dataflow import Dataflow
+
+brokers = ["localhost:19092"]
+flow = Dataflow("example")
+kinp = op.input("kafka-in", flow, KafkaSource(brokers, ["in-topic"]))
+processed = op.map("map", kinp, lambda x: KafkaSinkMessage(x.key, x.value))
+op.output("kafka-out", processed, KafkaSink(brokers, "out-topic"))
+```
+
+Or use the Kafka input [operator](/apidocs/bytewax.connectors/kafka/operators#bytewax.connectors.kafka.operators.input):
+
+```python
+from bytewax.connectors.kafka import operators as kop, KafkaSinkMessage
+from bytewax import operators as op
+from bytewax.dataflow import Dataflow
+
+brokers = ["localhost:19092"]
+flow = Dataflow("example")
+kinp = kop.input("kafka-in", flow, brokers=brokers, topics=["in-topic"])
+processed = op.map("map", kinp.oks, lambda x: KafkaSinkMessage(x.key, x.value))
+kop.output("kafka-out", processed, brokers=brokers, topic="out-topic")
+```
```

Comment:
In my head, the custom operator can offer some niceties that the source alone doesn't, but you can use the source if you don't want any of them. For example the `kop.input` operator returns the result splitted in 2 streams, and you have to manually raise on kafka error by inspecting the errors stream if you want your dataflow to crash on kafka errors. If you use the source with `op.input`, you can make it raise directly if an error occurs.

The `kop.output` offers the automatic convertion between `KafkaSourceMessage` and `KafkaSinkMessage`, because the sink implementation shouldn't bother to handle both, but the operator can add a map step before each message and do this. Here too, you might want to avoid this, and can use the sink directly